font/src/darwin/mod.rs

```diff
@@ -454,6 +454,33 @@ impl Font {
     }
 
     pub fn get_glyph(&self, character: char, _size: f64, use_thin_strokes: bool) -> Result<RasterizedGlyph, Error> {
+        // Render custom symbols for underline and beam cursor
+        match character {
+            super::UNDERLINE_CURSOR_CHAR => {
+                // Get the bottom of the bounding box
+                let descent = -(self.ct_font.descent() as i32);
+                // Get the width of the cell
+                let width = self.glyph_advance('0') as i32;
+                // Return the new custom glyph
+                return super::get_underline_cursor_glyph(descent, width);
+            },
+            super::BEAM_CURSOR_CHAR => {
+                // Get the top of the bounding box
+                let metrics = self.metrics();
+                let height = metrics.line_height;
+                let mut ascent = height - self.ct_font.descent() + 1.;
+                if ascent.floor() == ascent {
+                    // Fix off-by-one with an exact X.0 ascent
+                    ascent -= 1.;
+                }
```

Comment:
If the ascent is exactly XXX.0, it's gonna be off by one pixel and the bar will be offset to the top. Right now I've fixed this by simply checking if it is exactly .0, however there are probably a few different approaches to this.

Adding .9999... instead of 1. would probably also fix this, but is a rather unclean solution. Would love some feedback on te approach I've taken.

Reply:
Turns out this is the same as using `.ceil()`, could have thought of that earlier.